ffc/codegeneration/jit.py

```diff
@@ -298,9 +164,10 @@ def compile_coordinate_maps(meshes, parameters=None):
     cmap_names = [ffc.ir.representation.make_coordinate_mapping_jit_classname(
         mesh.ufl_coordinate_element(), "JIT", p) for mesh in meshes]
 
-    obj, mod = get_cached_module(module_name, cmap_names, p)
-    if obj is not None:
-        return obj, mod
+    if p['use_cache']:
```

Comment:
This doesn't really make sense when `p` has a default value of `None`. Will lead to runtime error.

The are other examples of this issue above.

Reply:
it doesn't crash because of the call to `validate_parameters()`, something I haven't changed. Not very clear code, I agree.

Reply:
Ok. I find `validate_parameters()` unclear, and in fact misleading. It doesn't just validate - it changes the parameter `dict`.

Reply:
Agree. It was there from before. I can make a change and add it to this PR.